### PR TITLE
Show better error when using ref cell instead of not

### PR DESCRIPTION
**Pull request by forki**
image

### CODE REVIEW

**Comment by dsyme** [comment]
Well that looks good :)

File: src/fsharp/CompileOps.fs [hunk 1]
```diff
@@ -654,9 +654,12 @@ let OutputPhasedErrorR (os:System.Text.StringBuilder) (err:PhasedError) =
           match contextInfo with
           | ContextInfo.OmittedElseBranch -> os.Append(FSComp.SR.missingElseBranch(t2)) |> ignore
           | ContextInfo.ElseBranch -> os.Append(FSComp.SR.elseBranchHasWrongType(t1,t2)) |> ignore
-          | ContextInfo.TupleInRecordFields -> 
+          | ContextInfo.TupleInRecordFields ->
+                os.Append(ErrorFromAddingTypeEquation1E().Format t2 t1 tpcs) |> ignore
+                os.Append(System.Environment.NewLine + FSComp.SR.commaInsteadOfSemicolonInRecord()) |> ignore
+          | _ when t2 = "bool" && t1.EndsWith " ref" ->
```
Comment (dsyme) [inline src/fsharp/CompileOps.fs]: A bit of a hacky way to get this condition, but I suppose it's good enough :)  It's hard to imagine it no working.

Comment (forki) [inline src/fsharp/CompileOps.fs]: yeah tried a couple of things, but this was still the "best"

**Comment by forki** [comment]
ready for inital review

**Comment by KevinRansom** [comment]
Looks good to me.

@forki Thanks for taking care of this.

Kevin

**Comment by cartermp** [comment]
👍

**Pull request merged**